packages/date-picker/src/vaadin-date-picker-overlay-content.js

```diff
@@ -855,15 +863,29 @@ class DatePickerOverlayContent extends ControllerMixin(ThemableMixin(DirMixin(Po
     if (!keepMonth) {
       this._focusedMonthDate = dateToFocus.getDate();
     }
-    await this.focusDateElement();
+    await this.focusDateElement(false);
   }
 
-  async focusDateElement() {
+  async focusDateElement(reveal = true) {
```

Comment:
From now on, the main difference between `focusDate()` and `focusDateElement()` as as follows:

- `focusDate()` should only be called when the focus is already in the calendar and focused date is changing (especially using keyboard navigation by arrow keys, <kbd>Page Up</kbd> and <kbd>Page Down</kbd> keys).
- `focusDateElement()` should be called when moving focus to the calendar, including cases when `focusedDate` is already set, but the actual `<td>` element that has `tabindex="0"` might be not in the viewport.